jmdaemon/jmdaemon/onionmc.py

```diff
@@ -1200,6 +1202,8 @@ def process_handshake(self, peerid: str, message: str,
                 return
             # We received a valid, accepting dn-handshake. Update the peer.
             peer.update_status(PEER_STATUS_HANDSHAKED)
+            # Show the info for the directory to the user, with emphasis:
+            self.info_callback("\n\n" + motd + "\n\n")

```

Comment:
We should limit allowed character set here, for example, don't allow ASCII control characters.

Reply:
Yes good point. Also there is the question of size. A slight concern I have is, it would be very natural to add e.g. gpg signatures here, but they would represent a big chunk of spam. I guess best practice will be, put a url here where the signature can be found.